jiant/tasks/lm.py

```diff
@@ -246,16 +246,26 @@ def update_metrics(self, out, batch=None):
         return
 
     def get_data_iter(self, path):
-        """Loading data file and tokenizing the text
+        """
+        Loading data file and tokenizing the text. We treat the Wikipedia corpus as a
+        long sequence, and we take each slice of 510 tokens as an example. The dataset
+        consists of a sentence per row in the file. This function concatenates all of the sentences,
+        before going through the sequence and yielding each chunk of 510 tokens.
         Args:
             path: (str) data file path
         """
+        seq_len = self.max_seq_len - 2
+        total_tokens = []
         with open(path, "r", encoding="utf-8") as txt_fh:
             for row in txt_fh:
                 toks = row.strip()
                 if not toks:
                     continue
-                yield tokenize_and_truncate(self._tokenizer_name, toks, self.max_seq_len)
+                toks = tokenize_and_truncate(self._tokenizer_name, toks, -1)
+                total_tokens += toks
```

Comment:
Are the datasets small enough that loading them into memory (`total_tokens += toks`) isn't a memory issue?

Reply:
It hasn't been an issue so far.

Reply:
Please update to something like the below to ease memory load:

```
                token_buffer += toks
                if len(token_buffer) > seq_len:
                    token_sequence = token_buffer[:seq_len]
                    token_buffer = token_buffer[seq_len:]
                    yield token_sequence
```

(`token_buffer` replaces your `total_tokens`, otherwise you should be able to drop this in, replacing L266-269)

Reply:
Done.